commands/utils.py

```diff
@@ -78,8 +80,24 @@ async def deploy(
                 ORDERED_ENVIRONMENTS,
             )
 
-        check_suites = await gh.get_check_suites(ref)
-        if not all(check["conclusion"] == "success" for check in check_suites if check["app"]["slug"] != "dependabot"):
+        # Collect check runs from all pages
+        check_runs, total_count = await gh.get_check_runs(ref, page=1)
+        for runs, _ in await asyncio.gather(
+            *(gh.get_check_runs(ref, page=page) for page in range(total_count // len(check_runs))),
```

Comment:
I think the current code is broken, e.g. for total_count = 50 you will do `for page in range(1)` or am I wrong?

What about this?
```suggestion
            total_pages = math.ceil(total_count / per_page)
            *(gh.get_check_runs(ref, page=page) for page in range(2, total_pages + 1)),
```
So sample cases:
- total_count=0 -> range(2, 2)
- total_count=50 -> range(2, 2)
- total_count = 100 -> range(2, 2)
- total_count = 101 -> range(2, 3)

Reply:
> I think the current code is broken, e.g. for total_count = 50 you will do `for page in range(1)` or am I wrong?

You're right, I messed up with the old version of the code. Fixed.